### PR TITLE
Implemented plugin dependencies.

### PR DESCRIPTION
**Pull request by gbin**
This is in 2 parts:

- [Core] Section, "DependsOn" key, a plugin can declare a dependency
  there.
- BotPlugin.get_plugin(name), a function to retrieve a dependency after
  activation.

### CODE REVIEW

**Comment by zoni** [comment]
What happens when two plugins depend on each other? So plugin A depends on plugin B, but plugin B also declares a dependency on plugin A.

**Comment by gbin** [comment]
Ha yeah, did not put the cyclic dependency detection in on purpose, gonna do that in another pull request.

File: errbot/plugin_manager.py [hunk 1]
```diff
@@ -272,7 +272,18 @@ def activate_plugin_with_version_check(self, plugin_info: PluginInfo) -> BotPlug
...
 
+        # Activate dependencies of this plugin first.
+        try:
+            depends_on = [dep_name.strip() for dep_name in plugin_info.details.get("Core", "DependsOn").split(',')]
+            for dep_name in depends_on:
+                if dep_name not in self.get_all_active_plugin_names():
+                    log.debug('%s depends on %s and %s is not activated. Activating it ...', name, dep_name, dep_name)
+                    self.activate_plugin(dep_name)
+        except NoOptionError:
+            depends_on = []
```
Comment (zoni) [inline errbot/plugin_manager.py]: `activate_plugin_with_version_check` is starting to get pretty long. Maybe move this section of code into a separate function to keep it a little more readable?

Comment (gbin) [inline errbot/plugin_manager.py]: done

**Comment by gbin** [comment]
Changed my mind :) one pull request with the circular dependency check included ;)

**Pull request merged**